mono/utils/mono-time.c

```diff
@@ -91,6 +91,35 @@ mono_100ns_datetime (void)
 
 #include <time.h>
 
+#if defined(HOST_DARWIN)
+static struct timeval time_at_boot = (struct timeval) {0};
+
+static struct timeval last_time = (struct timeval) {0};
+
+static const long max_delta_time_jump_before_cache_invalidation_us= 33000; // Span of one 30FPS frame
+
+inline void test_time_and_possibly_refresh_cache(struct timeval* now)
+{
+	struct timeval difference;
+
+	timersub(now, &last_time, &difference);
+
+	long diff_microseconds = (difference.tv_sec * 1e6) + difference.tv_usec;
```

Comment:
A few things about this line: 
`long` generally prefer a sized type with `int64_t`
`(difference.tv_sec * 1e6)` below in `mono_msec_boottime ` we upcast to `int64_t` before doing operations. Do we need that here?

Reply:
I've made the change you suggested.